static/src/javascripts/projects/common/modules/ui/first-pv-consent-plus-engagement-banner.js

```diff
@@ -123,13 +109,32 @@ const engagementMessage = new SubMessage(
 
 const show = (): Promise<boolean> => {
     trackFirstPvConsent();
-    if (document.body) {
-        document.body.insertAdjacentHTML('beforeend', bannerHtml);
-    }
-    bindFirstPvConsentClickHandlers(firstPvConsentMessage);
-    engagementMessage.bindCloseHandler(hideEngagementBanner);
-
-    return Promise.resolve(true);
+    return getEngagementBannerTestToRun()
+        .then(deriveEngagementBannerParams)
+        .then(getEngagementBannerHtml)
+        .then(engagementBannerHtml =>
+            fastdom.write(() => {
+                const html = doubleBannerHtml(engagementBannerHtml);
+                if (document.body) {
+                    document.body.insertAdjacentHTML('beforeend', html);
+                }
+                bindFirstPvConsentClickHandlers(firstPvConsentMessage);
+                engagementMessage.bindCloseHandler(hideEngagementBanner);
+                return true;
```

Comment:
oh, I should probably check whether the return value within the `write` callback actually becomes the resolved value of the fastdom promise. I suspect maybe it doesn't - which could break banner logic depending on the boolean result of `show()`

Reply:
did you check this?